pkg/encryption/cipher_test.go

```diff
@@ -51,12 +53,152 @@ func TestEncodeAndDecodeAccessTokenB64(t *testing.T) {
 	c, err := NewCipher([]byte(secret))
 	assert.Equal(t, nil, err)
 
-	encoded, err := c.Encrypt(token)
+	encoded, err := c.Encrypt([]byte(token))
 	assert.Equal(t, nil, err)
 
 	decoded, err := c.Decrypt(encoded)
 	assert.Equal(t, nil, err)
 
-	assert.NotEqual(t, token, encoded)
-	assert.Equal(t, token, decoded)
+	assert.NotEqual(t, []byte(token), encoded)
+	assert.Equal(t, []byte(token), decoded)
+}
+
+func TestEncryptAndDecrypt(t *testing.T) {
+	var err error
+
+	// Test our 3 cipher types
+	for _, initCipher := range []func([]byte) (Cipher, error){NewCipher, NewCFBCipher, NewGCMCipher} {
+		// Test all 3 valid AES sizes
+		for _, secretSize := range []int{16, 24, 32} {
+			secret := make([]byte, secretSize)
```

Comment:
One problem I have with tests like this, is it becomes hard to work out what went wrong, the whole test fails if one small part fails. While longer, for sake of clarity, this may be better as a set of subtests. Are you familiar with calling `t.Run("<name>", func(t *testing.T) {...})` within a test case? This way we could get individual subtests that fail and tell us what actually failed

Reply:
Thanks for the tip!  Added it 👍